caching/caching.go

```diff
@@ -50,29 +51,147 @@ func NewCachingMiddleware(mw *cachingMiddleWare) middleware.InitializeMiddleware
 			}
 			// Middleware ran successfully return our cached output
 			return o, middleware.Metadata{}, nil
+		case *dynamodb.BatchGetItemInput:
+			o, err := mw.handleBatchGetItemCommand(ctx, v, in, next)
+			if err != nil {
+				log.Printf("error occurred trying to use caching middleware. skipping middleware %+v\n", err)
+				return next.HandleInitialize(ctx, in) // continue request execution skip middleware
+			}
+			// Middleware ran successfully return our cached output
+			return o, middleware.Metadata{}, nil
 		}
 		// AWS SDK CMD not supported just continue as normal
 		return next.HandleInitialize(ctx, in)
 	})
 }
 
-func (d *cachingMiddleWare) handleGetItemCommand(ctx context.Context, input *dynamodb.GetItemInput, in middleware.InitializeInput, next middleware.InitializeHandler) (middleware.InitializeOutput, error) {
+func (d *cachingMiddleware) handleBatchGetItemCommand(ctx context.Context, input *dynamodb.BatchGetItemInput, in middleware.InitializeInput, next middleware.InitializeHandler) (middleware.InitializeOutput, error) {
+	responsesToReturn := make(map[string][]map[string]types.AttributeValue)
+	// for now any cache miss is considered a miss for the whole batch
+	gotMiss := false
+	// this holds the keys for each DDB table in the request and is used to compute the cache key for
+	// cache set operations
+	tableToKeys := make(map[string][]string)
+
+	for tableName, keys := range input.RequestItems {
+		// TODO: it may be preferable/safer to query the table for the keys
```

Comment:
@pgautier404 good call out. Im doing a DescribeTable call right now in seperate Async Writer im playing around w/ to determine what keys are since doesn't come back in scan response. I think way you are doing now though is good for first iteration lets keep this approach for now and can iterate if needed once we get some more milage and testing done.